rosidl_typesupport_c/cmake/get_used_typesupports.cmake

```diff
@@ -23,50 +29,43 @@
 # @public
 #
 function(get_used_typesupports var typesupport_type)
-  ament_index_get_resources(rmw_implementations "rmw_typesupport")
-  if(rmw_implementations STREQUAL "")
-    message(FATAL_ERROR "No RMW implementations found")
-  endif()
-
   # all type supports available
   ament_index_get_resources(available_typesupports "${typesupport_type}")
   if(available_typesupports STREQUAL "")
     message(FATAL_ERROR "No '${typesupport_type}' found")
   endif()
 
-  # supported type supports across all rmw implementations
-  set(supported_typesupports "")
-  foreach(rmw_implementation IN LISTS rmw_implementations)
-    ament_index_get_resource(resource "rmw_typesupport" "${rmw_implementation}")
-    foreach(ts IN LISTS resource)
-      if(NOT ts IN_LIST supported_typesupports)
-        list(APPEND supported_typesupports "${ts}")
+  # use explicitly provided list if provided
+  # option()
+  string(TOUPPER "${typesupport_type}" typesupport_type_upper)
+  if(NOT "$ENV{STATIC_${typesupport_type_upper}}" STREQUAL "")
+    string(REPLACE ":" ";" STATIC_${typesupport_type_upper} "$ENV{STATIC_${typesupport_type_upper}}")
```

Comment:
In this case, the ENV variable is overriding the compile option if provided?
Is that the expected behavior?

Reply:
It is the same behavior we are using for `RMW_IMPLEMENTATIONS`.